charts/opensearch/templates/statefulset.yaml

```diff
@@ -43,6 +43,9 @@ spec:
       storageClassName: "{{ .Values.persistence.storageClass }}"
     {{- end }}
     {{- end }}
+    {{- if .Values.persistence.selector }}
+    selector: "{{ .Values.persistence.selector }}"
```

Comment:
Hey @Pale-whale this will actually cause the following error when adding `matchLabels:` for selector in values.yaml
` error validating data: ValidationError(StatefulSet.spec.volumeClaimTemplates[0].spec.selector)`

I would suggest to test by adding in statefulset as follows.
```
{{- if .Values.persistence.matchLabels }}
  selector:
    matchLabels:
{{ toYaml .Values.persistence.pvc.matchLabels | indent 12 }}
{{- end}}
```

Reply:
@TheAlgo, @DandyDeveloper, @peterzhuamazon

Reply:
Good point